src/lib/network_pool/pool_sink.ml

```diff
@@ -158,7 +158,15 @@ module Base
         ( match cb' with
         | BC.External cb'' ->
             Diff.update_metrics env' cb''
-              (Option.some_if log_gossip_heard logger)
+              (Option.some_if log_gossip_heard logger) ;
+            don't_wait_for
+              ( match%map Mina_net2.Validation_callback.await cb'' with
+              | None ->
+                  [%log error]
+                    "Validation timed out on transaction/snark pool diff"
```

Comment:
It would be nice if you actually interpolate the "transaction" and "snark" strings into here. Then we get unique log strings to search for, making it easier to build a filter to count the validation timeouts by category.

Reply:
Also, can you reference `$diff` in the log message? It makes logproc output nicer.